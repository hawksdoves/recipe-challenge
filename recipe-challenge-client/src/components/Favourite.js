import React from 'react';
import { Button } from 'semantic-ui-react';

export default function Favourite(props) {
    const { addFavourite, recipeId, favourite, removeFavourite } = props

    return (
        favourite ?
            <Button color='red' icon='star' onClick={() => removeFavourite(recipeId)} /> :
            <Button color='blue' content='favourite me' onClick={() => addFavourite(recipeId)} />
        
    )
}
