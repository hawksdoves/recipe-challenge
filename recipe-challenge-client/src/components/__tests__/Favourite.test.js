import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'semantic-ui-react';

import Favourite from '../Favourite';

describe('Favourite', () => {
    it('should render text Un-Favourite Me when favourite', () =>{
        const wrapper = shallow(<Favourite favourite={true} />);
        
        expect(wrapper).toMatchSnapshot();
    })

    it('should render text Favourite Me when favourite', () =>{
        const wrapper = shallow(<Favourite />);

        expect(wrapper).toMatchSnapshot();
    })

    it('should call removeFavourite onclick ', () =>{
        const mockRemoveFavourite = jest.fn();
        const wrapper = shallow(<Favourite favourite={true} removeFavourite={mockRemoveFavourite} />);
        wrapper.find(Button).simulate('click');

        expect(mockRemoveFavourite).toHaveBeenCalled();
    })

    it('should call addFavourite onclick ', () =>{
        const mockAddFavourite = jest.fn();
        const wrapper = shallow(<Favourite addFavourite={mockAddFavourite} />);
        wrapper.find(Button).simulate('click');

        expect(mockAddFavourite).toHaveBeenCalled();
    })
});
