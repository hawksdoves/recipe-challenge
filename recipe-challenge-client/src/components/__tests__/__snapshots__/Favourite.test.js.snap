// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite should render text Favourite Me when favourite 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    as="button"
    color="blue"
    content="favourite me"
    onClick={[Function]}
/>,
  "nodes": Array [
    <Button
      as="button"
      color="blue"
      content="favourite me"
      onClick={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Favourite />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          as="button"
          color="blue"
          content="favourite me"
          onClick={[Function]}
/>,
        "_debugID": 4,
        "_renderedOutput": <Button
          as="button"
          color="blue"
          content="favourite me"
          onClick={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Favourite />,
}
`;

exports[`Favourite should render text Un-Favourite Me when favourite 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    as="button"
    color="red"
    icon="star"
    onClick={[Function]}
/>,
  "nodes": Array [
    <Button
      as="button"
      color="red"
      icon="star"
      onClick={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Favourite
        favourite={true}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "favourite": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          as="button"
          color="red"
          icon="star"
          onClick={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <Button
          as="button"
          color="red"
          icon="star"
          onClick={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Favourite
    favourite={true}
/>,
}
`;
